include/wabt/type-checker.h

```diff
@@ -172,13 +173,13 @@ class TypeChecker {
                            Type expected3,
                            const char* desc);
   Result CheckOpcode1(Opcode opcode,
-                      const Limits* limits = nullptr,
+                      std::optional<Type> type = std::nullopt,
```

Comment:
Maybe give this a more meaningful name?  How about `memory_type`?   Previously it was more obvious that the limits argument related to the memory, but `type` on it own in less clear I think.

Reply:
Yep, I changed them all to `memory_type`.